x-pack/platform/plugins/shared/alerting/server/routes/alert_delete/apis/schedule/post_alert_delete_schedule_route.ts

```diff
@@ -0,0 +1,92 @@
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { IRouter } from '@kbn/core/server';
+import { DEFAULT_SPACE_ID } from '@kbn/spaces-plugin/common/constants';
+import Boom from '@hapi/boom';
+import { alertDeleteScheduleQuerySchemaV1 } from '../../../../../common/routes/alert_delete';
+import type { ILicenseState } from '../../../../lib';
+import type { AlertingRequestHandlerContext } from '../../../../types';
+import { INTERNAL_BASE_ALERTING_API_PATH } from '../../../../types';
+import { verifyAccessAndContext } from '../../../lib';
+import { API_PRIVILEGES } from '../../../../../common';
+import { transformRequestToAlertDeleteScheduleV1 } from '../../transforms';
+
+export const alertDeleteScheduleRoute = (
+  router: IRouter<AlertingRequestHandlerContext>,
+  licenseState: ILicenseState
+) => {
+  router.post(
+    {
+      path: `${INTERNAL_BASE_ALERTING_API_PATH}/rules/settings/_alert_delete_schedule`,
+      validate: {
+        body: alertDeleteScheduleQuerySchemaV1,
+      },
+      security: {
+        authz: {
+          requiredPrivileges: [
+            `${API_PRIVILEGES.READ_ALERT_DELETE_SETTINGS}`,
+            `${API_PRIVILEGES.WRITE_ALERT_DELETE_SETTINGS}`,
+          ],
+        },
+      },
+      options: {
+        access: 'internal',
+      },
+    },
+    router.handleLegacyErrors(
+      verifyAccessAndContext(licenseState, async function (context, req, res) {
+        const alertingContext = await context.alerting;
+        const { spaceIds, ...settings } = transformRequestToAlertDeleteScheduleV1(req.body);
+
+        if (spaceIds && spaceIds.length > 0) {
+          const hasRequiredPrivilegeGranted =
+            await alertingContext.hasRequiredPrivilegeGrantedInAllSpaces({
+              request: req,
+              spaceIds,
+              requiredPrivilege: API_PRIVILEGES.WRITE_ALERT_DELETE_SETTINGS,
+            });
+
+          if (!hasRequiredPrivilegeGranted) {
+            throw Boom.forbidden(
+              'Insufficient privileges to delete alerts in the specified spaces'
+            );
+          }
+        }
+
+        if (!settings.isActiveAlertDeleteEnabled && !settings.isInactiveAlertDeleteEnabled) {
+          return res.badRequest({
+            body: {
+              message:
+                'active_alert_delete_threshold or inactive_alert_delete_threshold must be set',
+            },
+          });
+        }
+
+        try {
+          await alertingContext
+            .getAlertDeletionClient()
+            .scheduleTask(
+              req,
+              settings,
+              spaceIds && spaceIds.length > 0
+                ? spaceIds
+                : [(await alertingContext.getRulesClient()).getSpaceId() || DEFAULT_SPACE_ID]
+            );
+          return res.noContent();
+        } catch (error) {
+          return res.customError({
```

Comment:
Does the `throw Boom.forbidden` being caught here, and the `403` is returned as `500`?

Reply:
no, it does 403. I'll add a test